src/main/java/io/vertx/ext/mail/MailConfig.java

```diff
@@ -402,9 +388,15 @@ public String getKeyStorePassword() {
    *
    * @param keyStorePassword the key store passwords to be set
    * @return a reference to this, so the API can be used fluently
+   * @deprecated 4.0.0
    */
   public MailConfig setKeyStorePassword(String keyStorePassword) {
-    this.keyStorePassword = keyStorePassword;
+    JksOptions options = getTrustStoreOptions();
+    if (options == null) {
+      options = new JksOptions();
+      this.setTrustOptions(options);
+    }
+    options.setPassword(keyStorePassword);
     return this;
   }
 
```

Comment:
When we extends another `DataObject`, we need to override the `setter` methods to be able to keep it fluent. At least the `setProxyOptions` should be override in this case as it is the issue we want to address in the first place.